### PR TITLE
feat(suite): refactor the hash-check code to unify code

### PR DESCRIPTION
**Pull request by peter-sanderson**

### CODE REVIEW

File: docs/packages/connect/methods/getDeviceState.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ Get the state of a device.
 ```javascript
 const capturedDevice = await TrezorConnect.getDeviceState({
```
Comment (mroz22) [inline docs/packages/connect/methods/getDeviceState.md:20]: please split PR this to at least 2 commits. we use git log to generate connect changelog and we don't want to see there stuff like `feat(suite)` 🙏

**Review by komret — COMMENTED** [review]
Tested, looks good, but see @mroz22's comment.

File: docs/packages/connect/methods/getDeviceState.md (hunk 1 above)
Comment (peter-sanderson) [reply docs/packages/connect/methods/getDeviceState.md:20]: done

Comment (komret) [reply docs/packages/connect/methods/getDeviceState.md:20]: Should all connect code be in separate commits, @mroz22?

Comment (peter-sanderson) [reply docs/packages/connect/methods/getDeviceState.md:20]: Good point, I have split the all `connect` code.

**Review by komret — APPROVED** [review]

**Pull request merged**